src/js/edu-benefits/5490/config/form.js

```diff
@@ -498,11 +502,25 @@ const formConfig = {
                   'ui:options': {
                     hideIf: form => {
                       const status = _.get('highSchool.status', form);
-                      return status !== 'graduated';
+                      return status !== 'graduated' && status !== 'graduationExpected';
                     },
-                    expandUnder: 'status'
-                  }
-                }
+                    expandUnder: 'status',
+                    updateSchema: (pageData, form, schema) => {
+                      const status = _.get('educationHistory.data.highSchool.status', form);
+                      const field = _.get('educationHistory.uiSchema.highSchool.view:highSchoolOrGedCompletionDate', form);
+
+                      if (status === 'graduationExpected') {
+                        field['ui:title'] = 'When do you expect to earn your high school diploma?';
```

Comment:
I think you can do this instead for updateSchema:

```js
  const status = _.get('educationHistory.data.highSchool.status', form);

  if (status === 'graduationExpected') {
    return { title: 'When do you expect to earn your high school diploma?' };
  }
  
  return { title: 'When did you earn your high school diploma?' };
```

You'll have to set the first argument to date.uiSchema to `null`.